lisa/runners/lisa_runner.py

```diff
@@ -308,12 +309,16 @@ def _deploy_environment_task(
                     # rerun prepare to calculate resource again.
                     environment.status = EnvironmentStatus.New
         except Exception as identifier:
-            self._attach_failed_environment_to_result(
-                environment=environment,
-                result=test_results[0],
-                exception=identifier,
-            )
-            self._delete_environment_task(environment=environment, test_results=[])
+            if self._check_retry_limit():
```

Comment:
The name can be more clear like `_need_retry`. So it hints the output is true/false to retry or not.

Reply:
updated